### PR TITLE
Feature: sonic integration

### PR DESCRIPTION
**Pull request by akhlopiachyi**

### CODE REVIEW

**Review by oyershov — APPROVED** [review]

**Review by oyershov — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+$(APPS):
+	go build --ldflags "-X main.Version=$$(cat VERSION)" -o bin/$@ app.go
+
+test:
+	go test ./...
+
+asset:
+	cd web && \
+		yarn install && \
```
Comment (akushniruk) [inline Makefile:13]: add		```rm -rf ../public/assets && \
		rm -rf ./build && \```

File: handlers/index.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+	// Serve static files
+	router.Static("/public", "./public")
+
+	// React is looking for these files in root
+	// TODO: find solution for react build (webpack)
+	router.Static("/charting_library", "./public/assets/charting_library")
+
+	router.GET("/", index)
+	router.GET("/page", emptyPage)
```
Comment (akushniruk) [inline handlers/index.go:63]: remove this route

File: handlers/index.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+		"add": func(a int, b int) int {
+			return a + b
+		},
+	})
+}
+
+// render only file, must full name with extension
+func emptyPage(ctx *gin.Context) {
+	ctx.HTML(http.StatusOK, "page.html", gin.H{"title": "Page file title!!"})
+}
```
Comment (akushniruk) [inline handlers/index.go:129]: we do not need it

File: handlers/index.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	}
+
+	ctx.HTML(http.StatusOK, "index", gin.H{
+		"title":    "BaseAPP",
+		"cssFiles": cssFiles,
+		"jsFiles":  jsFiles,
+		"rootID":   "root",
+		"add": func(a int, b int) int {
+			return a + b
+		},
```
Comment (akushniruk) [inline handlers/index.go:122]: we do not need it

File: views/layouts/master.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        {{template "head" .}}
+        {{ if .cssFiles}}
+            {{range .cssFiles}}
+                <link href={{ . }} rel="stylesheet">
+            {{end}}
+        {{end}}
+    </head>
+
+    <body>
+        <div id={{ .rootID }}></div>
```
Comment (akushniruk) [inline views/layouts/master.html:16]: add config.js from public/assets
```
<script src="/public/assets/config.js"></script>
```
(https://github.com/openware/baseapp/blob/master/public/config.js)

**Review by akushniruk — CHANGES_REQUESTED** [review]

**Review by akushniruk — APPROVED** [review]

File: handlers/index.go (hunk 2 above)
Comment (akhlopiachyi) [reply handlers/index.go]: Done

File: handlers/index.go (hunk 3 above)
Comment (akhlopiachyi) [reply handlers/index.go]: Done

File: views/layouts/master.html (hunk 5 above)
Comment (akhlopiachyi) [reply views/layouts/master.html:17]: Done

**Pull request merged**